### PR TITLE
Fix #7428: Move shield and privacy settings to SwiftUI

### LINKED ISSUE #7428: Rewrite Shields popover UI in SwiftUI
### Description: 
The code is old and very limited by Static. Replacing this screen will make it easier to implement features such as standard/aggressive ad-blocking

### PR DESCRIPTION
**Pull request by cuba**
## Summary of Changes

This pull request fixes #7428

## Submitter Checklist:

- [ ] *Unit Tests* are updated to cover new or changed functionality
- [ ] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:

1. Need to go through all the toggles and links and ensure they all work as they did before
- [ ] The tracking and adblocking toggle is reflected on sites that don't have site specific changes
- [ ] Test toggles are remembered on both dismissing the setting screen and on application restarts
- [ ] Test that changing one toggle doesn't change another toggle
- [ ] Test that "Block all cookies" shows a confirmation and the setting is reverted on cancel
- [ ] Test that "Clear data now" shows a confirmation and nothing happens on cancel and that a loading appears on accept
- [ ] Test that "Block switch to app notices" is reflected on filter list screen and vice versa
- [ ] Test that "Block cookie notices" is reflected on filter list screen and vice versa
- [ ] Test that "Private browsing only" shows a confirmation and if cancelled the setting is reverted and nothing happens and when accepted a loading indicator appears and after its loaded non-private tabs are destroyed
- [ ] Test navigating to "Manage website data" and "Privacy hub" is ok on iPhone and iPad
- [ ] Test clear data toggles are remembered after clearing data

Note: You can test each individual feature such as amp, https upgrades, redirect tracking pages. But these tests will take a lot of time and the features themselves have not been changed. I simply only changed the UI part of the toggles. If you do test any of these features its more than likely that any issues are not new and should probably have its own ticket.

## Screenshots:
| Screenshot | Screenshot |
|-------------|------------|
| Simulator Screenshot - iPhone 14 Pro - 2023-05-10 at 18 51 50 | Simulator Screenshot - iPhone 14 Pro - 2023-05-10 at 18 52 18 |
| Simulator Screenshot - iPhone 14 Pro - 2023-05-10 at 18 52 07 | Simulator Screenshot - iPhone 14 Pro - 2023-05-10 at 18 52 04 |

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by cuba** [comment]
@iccub I will replace this PR with the following ones:
1. https://github.com/brave/brave-ios/pull/7354
2. https://github.com/brave/brave-ios/pull/7437 

The two will use swiftui for the advanced privacy settings but do not change the site specific settings. If this is acceptable I will close this PR and in the future will add the site specific settings

File: Sources/Brave/Frontend/Settings/SettingsViewController.swift [hunk 1]
```diff
@@ -134,10 +134,10 @@ class SettingsViewController: TableViewController {
   }
 
   // Do not use `sections` directly to access sections/rows. Use DataSource.sections instead.
-  private var sections: [Static.Section] {
+  @MainActor private func makeSections() -> [Static.Section] {
```
Comment (iccub) [inline Sources/Brave/Frontend/Settings/SettingsViewController.swift]: I don't think MainActor annotations are needed here.
UIViewController methods are main actors by default

File: Sources/Brave/Frontend/Settings/SettingsViewController.swift [hunk 2]
```diff
@@ -210,22 +210,42 @@ class SettingsViewController: TableViewController {
     return section
   }()
 
-  private lazy var featuresSection: Static.Section = {
+  @MainActor private func makeFeaturesSection() -> Static.Section {
```
Comment (iccub) [inline Sources/Brave/Frontend/Settings/SettingsViewController.swift]: Same as above but feel free to double check it

File: Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift [hunk 3]
```diff
@@ -459,20 +459,37 @@ extension BrowserViewController: TopToolbarDelegate {
...
+              assertionFailure()
+              return
+            }
+            
+            if isLoading, spinner == nil {
+              let newSpinner = SpinnerView()
+              newSpinner.present(on: view)
+              spinner = newSpinner
+            } else {
+              spinner?.dismiss()
```
Comment (iccub) [inline Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift:483]: Should set spinner to nil here perhaps?

File: Sources/Brave/Frontend/Settings/SettingsViewController.swift [hunk 4]
```diff
@@ -210,22 +210,42 @@ class SettingsViewController: TableViewController {
...
+                  assertionFailure()
+                  return
+                }
+                
+                if isLoading, spinner == nil {
+                  let newSpinner = SpinnerView()
+                  newSpinner.present(on: view)
+                  spinner = newSpinner
+                } else {
+                  spinner?.dismiss()
```
Comment (iccub) [inline Sources/Brave/Frontend/Settings/SettingsViewController.swift:239]: same as my other comment about setting it to nil after dismiss

File: Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift (hunk 3 above)
Comment (cuba) [reply Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift:483]: It's a weak variable so it shouldn't matter. but It doesn't hurt

File: Sources/Brave/Frontend/Settings/SettingsViewController.swift (hunk 2 above)
Comment (cuba) [reply Sources/Brave/Frontend/Settings/SettingsViewController.swift]: Done

File: Sources/Brave/Frontend/Settings/SettingsViewController.swift (hunk 4 above)
Comment (cuba) [reply Sources/Brave/Frontend/Settings/SettingsViewController.swift:239]: Done

File: Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift (hunk 3 above)
Comment (cuba) [reply Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift:483]: Done

**Review by iccub — APPROVED** [review]

**Pull request merged**